examples/asr/experimental/speech_to_text_aug.py

```diff
@@ -0,0 +1,243 @@
+            rates = list(range(0, 8))
+            rate = rates[random.randint(0, len(rates) - 1)]
+            _ = subprocess.check_output(
+                f"sox {orig_f.name} -V0 -C {rate} -t amr-nb - | sox -t amr-nb - -V0 -b 16 -r 16000 {transcoded_f.name}",
+                shell=True,
+            )
+        elif self._codecs[codec_ind] == "g711":
+            transcoded_f = NamedTemporaryFile(suffix="_g711.wav")
+            _ = subprocess.check_output(
+                f"sox {orig_f.name} -V0  -r 8000 -c 1 -e a-law {transcoded_f.name}", shell=True
+            )
+
+        new_data = AudioSegment.from_file(transcoded_f.name, target_sr=16000)
+        data._samples = new_data._samples[0 : data._samples.shape[0]]
+        return
+
+
+def get_augmentor_dict():
+    audio_augmentations = dict(
+        rir_noise_aug=dict(
+            prob=0.5,
+            rir_manifest_path="/data/datasets/freesound_20s/rir_noises_tarred/rir_tarred/tarred/tarred_audio_manifest.json",
+            rir_tar_filepaths="/data/datasets/freesound_20s/rir_noises_tarred/rir_tarred/tarred/audio_{0..1}.tar",
+            rir_prob=0.5,
+            noise_manifest_paths=[
+                "/data/datasets/freesound_20s/rir_noises_tarred/noises_20s_tarred/tarred_audio_manifest.json",
+                "/data/datasets/freesound_20s/rir_noises_tarred/noises_20s_tarred_8k/tarred_audio_manifest.json",
+            ],
+            noise_tar_filepaths=[
+                "/data/datasets/freesound_20s/rir_noises_tarred/noises_20s_tarred/audio_{0..63}.tar",
+                "/data/datasets/freesound_20s/rir_noises_tarred/noises_20s_tarred_8k/audio_{0..63}.tar",
+            ],
+            min_snr_db=[0, 0],
+            max_snr_db=[30, 30],
+            orig_sampling_rate=[16000, 8000],
+            bg_noise_manifest_paths=[
+                "/data/datasets/freesound_20s/rir_noises_tarred/noises_20s_tarred/tarred_audio_manifest.json",
+                "/data/datasets/freesound_20s/rir_noises_tarred/noises_20s_tarred_8k/tarred_audio_manifest.json",
+            ],
+            bg_noise_tar_filepaths=[
+                "/data/datasets/freesound_20s/rir_noises_tarred/noises_20s_tarred/audio_{0..63}.tar",
+                "/data/datasets/freesound_20s/rir_noises_tarred/noises_20s_tarred_8k/audio_{0..63}.tar",
+            ],
+            bg_min_snr_db=[10, 10],
+            bg_max_snr_db=[40, 40],
+            bg_orig_sampling_rate=[16000, 8000],
+        ),
+        transcode_aug=dict(prob=0.5,),
+    )
+    return audio_augmentations
+
+
+@hydra_runner(config_path="../conf", config_name="quartznet_15x5")
+def main(cfg):
+    perturb.register_perturbation(name='rir_noise_aug', perturbation=RirAndNoisePerturbation)
+    perturb.register_perturbation(name='transcode_aug', perturbation=TranscodePerturbation)
+
+    OmegaConf.set_struct(cfg, False)
+    if cfg.trainer.gpus > 0 and "num_workers" not in cfg.model.train_ds:
+        cfg.model.train_ds.num_workers = os.cpu_count()
```

Comment:
@okuchaiev Let's mark this as a an automation goal for the future (with balancing for multi node etc)